core/templates/domain/skill/editable-skill-backend-api.service.ts

```diff
@@ -16,124 +16,159 @@
-        change_dicts: changeList
-      };
+    const putData = {
+      version: skillVersion,
+      commit_message: commitMessage,
+      change_dicts: changeList
+    };
 
-      $http.put(editableSkillDataUrl, putData).then(function(response) {
-        // The returned data is an updated skill dict.
-        var skill = angular.copy(response.data.skill);
-        if (successCallback) {
-          successCallback(skill);
-        }
-      }, function(errorResponse) {
+    this.http.put(
+      editableSkillDataUrl, putData
+    ).toPromise().then(
+      (data: editableSkillResponseConfig) => successCallback(data.skill),
+      error => {
         if (errorCallback) {
-          errorCallback(errorResponse.data);
+          errorCallback(error.error);
         }
-      });
-    };
-
-    var _deleteSkill = function(skillId, successCallback, errorCallback) {
-      var skillDataUrl = UrlInterpolationService.interpolateUrl(
-        EDITABLE_SKILL_DATA_URL_TEMPLATE, {
-          skill_id: skillId
-        });
+      }
+    );
+  }
 
-      $http['delete'](skillDataUrl).then(function(response) {
-        if (successCallback) {
-          successCallback(response.status);
-        }
-      }, function(errorResponse) {
+  // TODO(#7165): Replace any with exact type.
+  _deleteSkill(
+      skillId: string,
+      successCallback: (value?: Object | PromiseLike<Object>) => void,
+      errorCallback: (reason?: any) => void
+  ): void {
+    const skillDataUrl = this.urlInterpolationService.interpolateUrl(
+      SkillDomainConstants.EDITABLE_SKILL_DATA_URL_TEMPLATE, {
+        skill_id: skillId
+      });
+    this.http['delete'](skillDataUrl, { observe: 'response' }).toPromise().then(
+      (res) => successCallback(res.status),
+      (error) => {
         if (errorCallback) {
-          errorCallback(errorResponse.data);
+          errorCallback(error.error);
         }
-      });
-    };
-
-    return {
-      fetchSkill: function(skillId) {
-        return $q(function(resolve, reject) {
-          _fetchSkill(skillId, resolve, reject);
-        });
-      },
-      fetchMultiSkills: function(skillIds) {
-        return $q(function(resolve, reject) {
-          _fetchMultiSkills(skillIds, resolve, reject);
-        });
-      },
-      updateSkill: function(
-          skillId, skillVersion, commitMessage, changeList) {
-        return $q(function(resolve, reject) {
-          _updateSkill(
-            skillId, skillVersion, commitMessage, changeList,
-            resolve, reject);
-        });
-      },
-      deleteSkill: function(skillId) {
-        return $q(function(resolve, reject) {
-          _deleteSkill(skillId, resolve, reject);
-        });
       }
-    };
+    );
   }
-]);
+
+  fetchSkill(skillId: string): Promise<Object> {
+    return new Promise((resolve, reject) => {
+      this._fetchSkill(skillId, resolve, reject);
+    });
+  }
+
+  fetchMultiSkills(skillIds: [string]): Promise<Object> {
```

Comment:
Can we change [string] to Array<string>?

Reply:
I am sorry I don't get this? Do you want it to be `skillIds: Array`?

Reply:
Array<string>

Reply:
Ok done!